share/spice/currency/currency.js

```diff
@@ -286,14 +278,17 @@
                 $left_select.val(from);
             }
 
-            var endpoint = "/js/spice/currency/1/" + from + "/" + to;
-            $.get(endpoint, function(payload) {
-
-                // jsonp is returned from the API so we have to alter the contents
+            $.ajax({
+                url: "/js/spice/currency/1/" + from + "/" + to,
+                beforeSend: function( xhr ) {
+                    xhr.overrideMimeType( "text/plain; charset=x-user-defined" );
```

Comment:
Ah, I see. I think avoiding these work around with a new endpoint is a simpler solution. We can make alt to called `currency_json` and disable the jsonp wrapping.

Then you can `$.getJSON('/js/spice/currency_json/' ... `